### ISSUE TITLE
Flag illegal use of `ClassVar` in a type alias definition

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3412]
The following usage of `ClassVar` should be considered illegal, but it is not currently flagged as such.

```python
from typing import TypeAlias, ClassVar

ClassAlias: TypeAlias = ClassVar[str]
```

See this discussion: https://github.com/python/typing/discussions/1172

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.243, which I just published. It will also be included in the next release of pylance.
